x-pack/plugins/ingest_manager/public/applications/ingest_manager/hooks/use_intra_app_state.tsx

```diff
@@ -57,9 +57,13 @@ export function useIntraAppState<S = AnyIntraAppRouteState>():
     // once so that it does not impact navigation to the page from within the
     // ingest app. side affect is that the browser back button would not work
     // consistently either.
-    if (location.pathname === intraAppState.forRoute && !wasHandled.has(intraAppState)) {
+
+    if (
+      location.pathname + location.search === intraAppState.forRoute &&
```

Comment:
Looking for feedback on this, I'm thinking when we need search params, we pass them in separately and then keep this comparison to be by `pathname` only

Reply:
I agree that this should not trigger based off of `.search` string - only `pathname`. The page component is the one  that should handle retrieving any URL `.search` params and process on them.